make-bm-worker/README.md

```diff
@@ -0,0 +1,68 @@
+Registering Bare Metal Hosts
+============================
+
+The `make-bm-worker` tool may be a more convenient way of creating
+YAML definitions for workers than editing the files directly.
+
+```
+$ go run make-bm-worker/main.go -address 1.2.3.4 -password password -user admin worker-99
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: worker-99-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+apiVersion: metalkube.org/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: worker-99
+spec:
+  online: true
+  bmc:
+    address: 1.2.3.4
+    credentialsName: worker-99-bmc-secret
+```
+
+The output can be passed directly to `oc apply` like this:
+
+```
+$ go run make-bm-worker/main.go -address 1.2.3.4 -password password -user admin worker-99 | oc apply -f -
```

Comment:
I wonder if we should make the examples more cut/paste friendly for the default virt environment?

E.g I've done:

```
$ go run make-bm-worker/main.go -address ipmi://192.168.111.1:6233 -password password -user admin -image worker-0 | oc --config ocp/auth/kubeconfig apply -f -
secret/worker-0-bmc-secret created
baremetalhost.metalkube.org/worker-0 created
```

That's not actually working yet with the demo operator running, but it'd be nice to have an out-of-the box example given the scope of this repo?

Reply:
When you say "demo operator" do you mean the baremetal operator with the -demo-mode flag? Because that iteration doesn't actually provision, it looks for hosts with specific names to move them all into the various states so the UX team can see what they look like.

Reply:
Ah yeah I'd missed that, I was running `demo/launch_demo_bmo.sh` and expecting a worker to get built, I guess we can add a variable that allows that in a future PR, or copy/paste the necessary stuff into a `0N_deploy_workers.sh` pending fully integrated worker deployment